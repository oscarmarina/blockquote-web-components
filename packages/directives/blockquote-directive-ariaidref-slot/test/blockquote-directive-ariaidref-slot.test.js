/* eslint-disable import/no-extraneous-dependencies */
import {suite, test, assert, expect, beforeAll} from 'vitest';
import {fixture, fixtureCleanup} from '@open-wc/testing-helpers';
import {getDiffableHTML} from '@open-wc/semantic-dom-diff/get-diffable-html.js';
import {html} from 'lit';

import './blockquote-directive-ariaidref-slot-component.js';

suite('BlockquoteDirectiveAriaidrefSlot', () => {
  let el;
  let elShadowRoot;

  suite('Slot is hidden', () => {
    beforeAll(async () => {
      el = await fixture(html`
        <slot-idref-hidden></slot-idref-hidden>
      `);
      elShadowRoot = el?.shadowRoot?.innerHTML;

      return () => {
        fixtureCleanup();
      };
    });

    suite('Semantic Dom', () => {
      test('SHADOW DOM - Structure test', async () => {
        expect(getDiffableHTML(elShadowRoot)).toMatchSnapshot('SHADOW DOM');
      });
    });
  });

  suite('Slot is visible', () => {
    beforeAll(async () => {
      el = await fixture(html`
        <slot-idref-visible></slot-idref-visible>
      `);

      elShadowRoot = el?.shadowRoot?.innerHTML;

      return () => {
        fixtureCleanup();
      };
    });

    suite('Semantic Dom', () => {
      test('SHADOW DOM - Structure test', async () => {
        expect(getDiffableHTML(elShadowRoot)).toMatchSnapshot('SHADOW DOM');
      });
    });
  });

  suite('Thrown Error', () => {
    test('Error', async () => {
      try {
        await fixture(html`
          <slot-idref-error></slot-idref-error>
        `);
        assert.fail('Expected an error to be thrown');
      } catch (error) {
        assert.instanceOf(error, Error);
      }
    });
  });

  suite('No render', () => {
    beforeAll(async () => {
      el = await fixture(html`
        <slot-idref-nothing></slot-idref-nothing>
      `);
      elShadowRoot = el?.shadowRoot?.innerHTML;

      return () => {
        fixtureCleanup();
      };
    });

    suite('Semantic Dom', () => {
      test('SHADOW DOM - Structure test', async () => {
        expect(getDiffableHTML(elShadowRoot)).toMatchSnapshot('SHADOW DOM');
      });
    });
  });
});
