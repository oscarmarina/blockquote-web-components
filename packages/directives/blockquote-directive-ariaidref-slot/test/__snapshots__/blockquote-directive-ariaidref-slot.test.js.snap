// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BlockquoteDirectiveAriaidrefSlot > No render > Semantic Dom > SHADOW DOM - Structure test > SHADOW DOM 1`] = `
"<slot>
</slot>
<hr>
"
`;

exports[`BlockquoteDirectiveAriaidrefSlot > No render > Semantic Dom > SHADOW DOM - Structure test 1`] = `
"<slot>
</slot>
<hr>
"
`;

exports[`BlockquoteDirectiveAriaidrefSlot > Slot is hidden > Semantic Dom > SHADOW DOM - Structure test > SHADOW DOM 1`] = `
"<slot>
</slot>
<hr>
<slot
  hidden=""
  id="hi"
  name="hi"
>
</slot>
"
`;

exports[`BlockquoteDirectiveAriaidrefSlot > Slot is hidden > Semantic Dom > SHADOW DOM - Structure test 1`] = `
"<slot>
</slot>
<hr>
<slot
  hidden=""
  id="hi"
  name="hi"
>
</slot>
"
`;

exports[`BlockquoteDirectiveAriaidrefSlot > Slot is visible > Semantic Dom > SHADOW DOM - Structure test > SHADOW DOM 1`] = `
"<slot>
</slot>
<hr>
<slot
  id="hi"
  name="hi"
>
</slot>
"
`;

exports[`BlockquoteDirectiveAriaidrefSlot > Slot is visible > Semantic Dom > SHADOW DOM - Structure test 1`] = `
"<slot>
</slot>
<hr>
<slot
  id="hi"
  name="hi"
>
</slot>
"
`;
